### ISSUE TITLE
proxy not serving for get request

### ISSUE THREAD

**Issue by tripatabhishek** [chimurai/http-proxy-middleware#144]
Following is my configuration. Every request is being served by proxy except the GET type. 

Setup : 
Path: **server/index.js** 

import express from 'express';
import path from 'path';
import webpack from 'webpack';
import webpackMiddleware from 'webpack-dev-middleware';
import webpackHotMiddleware from 'webpack-hot-middleware';
import webpackConfig from '../webpack.config.dev';
let proxy = require('http-proxy-middleware');
let app = express();
const compiler = webpack(webpackConfig);
app.use(webpackMiddleware(compiler, {
    hot: true,
    noInfo: true
}));
app.use(webpackHotMiddleware(compiler));
app.get('/*', (req, res) => {
    res.sendFile(path.join(__dirname, './index.html'));
})
let options = {
    target: 'https://mysite.com',
    changeOrigin: true,
    logLevel:'debug',
    ws: true,
    onProxyReq: function(proxyReq, req, res) {
        //  log the req
        console.log('RAW REQUEST from the target', JSON.stringify(req.headers, true, 2));
    },
    onProxyRes: function (proxyRes, req, res) {
        //  log the response
        console.log('RAW Response from the target', JSON.stringify(proxyRes.headers, true, 2));
    }
};
app.use(['/front_end_api/**'], proxy(options));
app.listen(3000, () => console.log('Running on localhost'));

==========================================================

Please let me know if I am doing anything wrong here.

**Comment by chimurai** [comment]
Please use StackOverflow. I think that would be a better place to troubleshoot your setup.

Thanks.

**Comment by tripatabhishek** [comment]
Just posted. Let me know if you can help. 
http://stackoverflow.com/questions/42232511

**Comment by chimurai** [comment]
Think you are mixing up [Express Route paths](https://expressjs.com/en/guide/routing.html) with [HPM wildcard context matching](https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/context-matching.md#multi-wildcard).

Check https://www.npmjs.com/package/path-to-regexp for Express Route paths patterns.
Handy tester for Express Rout paths: http://forbeslindesay.github.io/express-route-tester/


`app.use(['/front_end_api/**'], proxy(options));  // faulty config`

expecting something like this:
- `app.use('/front_end_api', proxy(options));`
- `app.use(['/front_end_api'], proxy(options));`
- `app.use(proxy(['/front_end_api/**'], options));`
- `app.use(proxy(['/front_end_api'], options));`

**Comment by tripatabhishek** [comment]
Thanks, @chimurai for your quick and valuable response. 
However, there was a small mistake I did, that is I had to remove the wildcard '*' in the below code.

app.get('/*', (req, res) => {
    res.sendFile(path.join(__dirname, './index.html'));
})

It was using express routing instead of HPM middleware for all the get requests. now with the wildcard character(*) removed the get request with my API endpoints are being served properly. 

Thanks, Again.
